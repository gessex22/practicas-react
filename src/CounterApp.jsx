import { useState } from "react";
import PropTypes from "prop-types";

export  const CounterApp = ({value = 20}) => {
  const [counter, setCounter] = useState(value); // El estado inicial es `value`

  const handleAdd = () => setCounter(counter + 1);
  const handleMinus = () => setCounter(counter - 1);
  const handleReset = () => setCounter(value); // Reinicia al valor inicial (value)

  return (
    <>
      <h1>Counter App</h1>
      <h2>{counter}</h2>
      <button onClick={handleAdd}>+1</button>
      <button onClick={handleMinus}>-1</button>
      <button aria-label="btn-reset" onClick={handleReset}>Reset</button>
    </>
  );
};

// Definici√≥n de PropTypes (opcional pero recomendado)
CounterApp.propTypes = {
  value: PropTypes.number,
};

// Valor por defecto si no se pasa `value`
//CounterApp.defaultProps = {
  //value: 20, // Puedes cambiarlo a 20 si prefieres
//};