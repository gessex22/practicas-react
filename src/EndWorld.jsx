import PropTypes from "prop-types";

export const NdelWorld = ({ title, time }) => {
  return (
    <>
      <h1>{title} </h1>
      <p> Fin del mundo desde un modulo exportado</p>
      <p> {time + 1}</p>
    </>
  );
}

NdelWorld.propType = {
title: PropTypes.string.isRequired,
time: PropTypes.number.isRequired

}


NdelWorld.defaultProps = {

title : 'not title',
time: 99999
}





