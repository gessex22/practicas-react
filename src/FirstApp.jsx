import PropTypes from "prop-types"


export const FirstApp = ({title,subtitle, name})=> {


    return(
    
        <>
            <h1 data-testid="test-title">{ title}</h1>
            <p>{subtitle}</p>
             <p>{subtitle}</p>
            <p>{name}</p>
        
        </>
    
    
    )


}

FirstApp.propTypes = {

    name : PropTypes.string.isRequired
}


