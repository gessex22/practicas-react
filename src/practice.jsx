import PropTypes from "prop-types"


export const Practice = ({sentence})=> {

    const sumaDom = (event) =>{

    console.log( event)
    sentence = 1000
    
    }


    return(
    
        <>
            <h1>Practice </h1>
            <h2> {sentence }</h2>
            <button onClick={ sumaDom} > sumn </button>
        
        </>
    
    
    )


}

Practice.propTypes = {

    sentence : PropTypes.string
}


