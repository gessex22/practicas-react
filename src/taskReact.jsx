import PropTypes from "prop-types";

export const Task = ({ value }) => {
  return (
    <>
      <h1> CounterAPP</h1>
      <h2> {value}</h2>
    </>
  );
};

Task.propTypes = {
  value: PropTypes.number,
};
