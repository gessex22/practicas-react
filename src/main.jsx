import React from 'react';
import ReactDOM from 'react-dom/client';
// import { NdelWorld } from './EndWorld';

import './main.css'
import { CounterApp } from './CounterApp'
//import { Practice } from './practice';
//import { func } from 'prop-types';

 

// function EndWorl() {

//  return (<h1>EL verdader N del world</h1>);

// }


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode> 
     < CounterApp /> 
    </React.StrictMode>
)