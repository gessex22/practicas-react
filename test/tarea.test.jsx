import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe(" Tarea en el counter app", () => {
  test(" Match con el snapch", () => {
    expect(render(<CounterApp value={10} />).container).toMatchSnapshot();
  });

  test(" Debe iniciar con un 100 el counter", () => {
    render(<CounterApp value={100} />);
    expect(screen.getByText(100)).toBeTruthy();
  });

  test('debe de incrementar con el boton +1' , ()=>{
  
    render( <CounterApp />)
    fireEvent.click(screen.getByText('+1'))
    expect(screen.getByText('21')).toBeTruthy()
  
  })
  test('debe de decrementar con el boton -1' , ()=>{
  
    render( <CounterApp />)
    fireEvent.click(screen.getByText('-1'))
    expect(screen.getByText('19')).toBeTruthy()
  
  })
   test('debe de funcionar el boton de reset con el boton' , ()=>{
  
    render( <CounterApp value={28909} />)
    fireEvent.click(screen.getByText('+1'))
    fireEvent.click(screen.getByText('+1'))
    fireEvent.click(screen.getByRole ('button', {name: 'btn-reset'}))
    expect(screen.getByText (28909)).toBeTruthy()
  
  })
});
