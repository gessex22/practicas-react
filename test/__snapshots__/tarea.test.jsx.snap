// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Tarea en el counter app  Match con el snapch 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    10
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
</div>
`;
